lib/ansible/modules/storage/netapp/na_ontap_service_processor_network.py

```diff
@@ -0,0 +1,264 @@
+                'netmask_value': netmask_value,
+                'prefix_length_value': prefix_length_value
+            }
+        return sp_network_details
+
+    def modify_service_processor_network(self):
+        """
+        Modify a service processor network
+        """
+        service_obj = netapp_utils.zapi.NaElement(
+            'service-processor-network-modify')
+        service_obj.add_new_child("node", self.node)
+        service_obj.add_new_child("address-type", self.address_type)
+        service_obj.add_new_child("is-enabled", self.is_enabled)
+
+        if self.dhcp:
+            service_obj.add_new_child("dhcp", self.dhcp)
+        if self.gateway_ip_address:
+            service_obj.add_new_child(
+                "gateway-ip-address", self.gateway_ip_address)
+        if self.ip_address:
+            service_obj.add_new_child("ip-address", self.ip_address)
+        if self.netmask:
+            service_obj.add_new_child("netmask", self.netmask)
+        if self.prefix_length:
+            service_obj.add_new_child("prefix-length", str(self.prefix_length))
+
+        try:
+            result = self.server.invoke_successfully(service_obj,
+                                                     enable_tunneling=True)
+        except netapp_utils.zapi.NaApiError as error:
+            self.module.fail_json(msg='Error modifying \
+                                  service processor network: %s'
+                                  % (to_native(error)),
+                                  exception=traceback.format_exc())
+
+    def apply(self):
+        """
+        Run Module based on play book
+        """
+        changed = False
+        results = netapp_utils.get_cserver(self.server)
+        cserver = netapp_utils.setup_ontap_zapi(
+            module=self.module, vserver=results)
+        netapp_utils.ems_log_event(
+            "na_ontap_service_processor_network", cserver)
+        spn_details = self.get_service_processor_network()
+        spn_exists = False
+        if spn_details:
+            spn_exists = True
+            if self.state == 'present':  # modify
+                if (self.dhcp and
+                    self.dhcp != spn_details['dhcp_value']) or \
+                   (self.gateway_ip_address and
+                    self.gateway_ip_address != spn_details['gateway_ip_address_value']) or \
+                   (self.ip_address and
+                    self.ip_address != spn_details['ip_address_value']) or \
+                   (self.netmask and
+                    self.netmask != spn_details['netmask_value']) or \
+                   (self.prefix_length and str(self.prefix_length)
+                        != spn_details['prefix_length_value']):
+                    changed = True
+        else:
+            pass
+        if changed:
+            if self.module.check_mode:
+                pass
+            else:
+                if self.state == 'present':  # execute modify
+                    if spn_exists:
```

Comment:
definitely seems like there should be an else with an error here...

Reply:
We will address this in the next release.